runtime/src/bank.rs

```diff
@@ -2211,11 +2224,24 @@ impl Bank {
                     );
 
                     if enable_log_recording {
-                        transaction_logs.push(
+                        let mut log_messages: TransactionLogMessages =
                             Rc::try_unwrap(log_collector.unwrap_or_default())
                                 .unwrap_or_default()
-                                .into(),
+                                .into();
+
+                        let line_count = log_messages.len();
+
+                        Self::truncate_log_messages(
+                            &mut log_messages,
+                            TRANSACTION_LOG_MESSAGES_BYTES_LIMIT,
                         );
+
+                        if log_messages.len() != line_count {
```

Comment:
It would be nice if this were inside `truncate_log_messages` or similar helper so that you could unit-test this branch.

Reply:
I agree. I'll refactor that bit.